wandb/sdk/system/assets/tpu.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import multiprocessing as mp
+import os
+from collections import deque
+from typing import TYPE_CHECKING, List, Optional
+
+from wandb.sdk.system.assets.aggregators import aggregate_mean
+from wandb.sdk.system.assets.asset_registry import asset_registry
+from wandb.sdk.system.assets.interfaces import (
+    Interface,
+    Metric,
+    MetricsMonitor,
+    MetricType,
+)
+
+if TYPE_CHECKING:
+    from typing import Deque
+
+    from wandb.sdk.internal.settings_static import SettingsStatic
+
+try:
+    from tensorflow.python.distribute.cluster_resolver import (  # type: ignore
+        tpu_cluster_resolver,
+    )
+    from tensorflow.python.profiler import profiler_client  # type: ignore
+except (
+    ImportError,
+    TypeError,
+    AttributeError,
+):  # Saw type error when iterating paths on colab...
+    # TODO: Saw error in sentry where module 'tensorflow.python.pywrap_tensorflow'
+    #  has no attribute 'TFE_DEVICE_PLACEMENT_EXPLICIT'
+    tpu_cluster_resolver = None
+    profiler_client = None
+
```

Comment:
I changed this file maybe this caused the hang

Reply:
ok, reverted. let's see if it helped :)